pyro/infer/mcmc/nuts.py

```diff
@@ -30,8 +28,8 @@ def _logaddexp(x, y):
 # weight is the number of valid points in case we use slice sampling
 #   and is the log sum of (unnormalized) probabilites of valid points
 #   when we use multinomial sampling
-_TreeInfo = namedtuple("TreeInfo", ["z_left", "r_left", "z_left_grads",
-                                    "z_right", "r_right", "z_right_grads",
+_TreeInfo = namedtuple("TreeInfo", ["z_left", "r_left", "r_left_unscaled", "z_left_grads",
+                                    "z_right", "r_right", "r_right_unscaled", "z_right_grads",
```

Comment:
We don't want to recompute those unscaled terms. So I expand the TreeInfo to store that extra information.